noxfile.py

```diff
@@ -0,0 +1,263 @@
+            if f.is_dir():
+                shutil.rmtree(f)
+            else:
+                f.unlink()
+
+
+# UTILITY FUNCTIONS
+###############################################################################
+
+
+def install_requirements(session, requirements, package_manager="pip"):
+    """
+    Install dependencies from the requirements files specified in REQUIREMENTS.
+
+    *requirements* should be a list of keywords defined in the REQUIREMENTS
+    dictionary. Set *package_manager* to "conda" if using the conda backend for
+    virtual environments.
+    """
+    if package_manager not in {"pip", "conda"}:
+        raise ValueError(f"Invalid package manager '{package_manager}'")
+    if session.posargs and "skip-install" in session.posargs:
+        session.log(f"Skipping install steps.")
+        return
+    arg_name = {"pip": "-r", "conda": "--file"}
+    args = []
+    for requirement in requirements:
+        args.extend([arg_name[package_manager], REQUIREMENTS[requirement]])
+    if package_manager == "pip":
+        session.install(*args)
+    elif package_manager == "conda":
+        session.conda_install(
+            "--channel=conda-forge",
+            "--channel=defaults",
+            *args,
+        )
+
+
+def list_packages(session, package_manager="pip"):
+    """
+    List installed packages in the virtual environment.
+
+    If the argument 'list-packages' is passed to the nox session, will list the
+    installed packages in the virtual environment (using the package manager
+    specified).
+
+    Example: 'nox -s test-3.7 -- list-packages'
+    """
+    if package_manager not in {"pip", "conda"}:
+        raise ValueError(f"Invalid package manager '{package_manager}'")
+    if session.posargs and "list-packages" in session.posargs:
+        session.log(f"Packages installed ({package_manager}):")
+        if package_manager == "pip":
+            session.run("pip", "freeze")
+        elif package_manager == "conda":
+            session.run("conda", "list")
+
+
+def run_pytest(session):
+    """
+    Run tests with pytest in a separate folder.
+    """
+    tmpdir = session.create_tmp()
+    session.cd(tmpdir)
+    session.run("pytest", *PYTEST_ARGS, PACKAGE, *RST_FILES)
+    session.run("coverage", "xml", "-o", ".coverage.xml")
+    # Copy the coverage information back so it can be reported
+    for f in Path(".").glob(".coverage*"):
+        shutil.copy(f, Path(__file__).parent)
+
+
+def build_project(session, install=False):
+    """
+    Build source and wheel packages for the project and returns their path.
+    If 'install==True', will also install the package.
+    """
+    session.log("Build source and wheel distributions:")
+    if Path("dist").exists():
+        shutil.rmtree("dist")
+    session.run("python", "setup.py", "sdist", "bdist_wheel", silent=True)
+    packages = list(Path("dist").glob("*"))
+    if install:
+        session.install("--force-reinstall", "--no-deps", str(packages[0]))
```

Comment:
I'm getting an error when running `nox --install-only` because `packages` is empty.
Maybe we can check if `packages` is not empty before installing the package.
```suggestion
    if install and packages:
        session.install("--force-reinstall", "--no-deps", str(packages[0]))
```